.travis.yml

```diff
@@ -46,9 +50,8 @@ before_install:
   - cargo-audit -V || cargo install cargo-audit --force
     # List all installed cargo packages.
   - cargo install --list
-    # libjava_bindings.so and native application both need to load common Rust libs (eg libstd.so).
-  - export LD_LIBRARY_PATH=$RUST_LIB_DIR:$LD_LIBRARY_PATH
-    # force building instead of using from apt.
+    # Force building Exonum dependencies instead of using the system ones, if any.
+    # Required because using system results in broken artefacts on older CPUs, see ECR-1658.
   - export SODIUM_BUILD=1
```

Comment:
If we decided to install system libsodium.so for Java, should we reuse it in Rust part too?
`export SODIUM_LIB_DIR="/usr/lib"`

Reply:
We had crashes with libsodium from repos, remember?

Reply:
I noticed that we build libsodium but couldn't remember _why_ we do. Still can't remember the crashes :man_shrugging: 

#399 might make system libsodium unnecessary

Reply:
afaik we had some problems (not exactly crashes), so @alexander-irbis found that building libsodium from source solves these problems.

Reply:
I remember RocksDB & SIGILL on some older CPUs.

Reply:
:thinking: interesting, so why we build libsodium?

Reply:
> We had crashes with libsodium from repos, remember?

Definitely that issue with SIGILL was discussed primarily in the context of RocksDB.
LIbsodium was switched to build at the same time with RocksDB but for another reason, as try to fix crashing issue on Linux. It did not help.

Reply:
I see, thanks! So we could remove that line and have exonum pick up the system one?

Reply:
Updated and cleared the caches on Travis to get a clean build, will see how it goes: https://travis-ci.org/exonum/exonum-java-binding/builds/411277584